infrastructure/lib/constructs/compliant-resources.ts

```diff
@@ -1,18 +1,29 @@
 import * as s3 from "@aws-cdk/aws-s3";
 import * as cdk from "@aws-cdk/core";
 
-/**
- * Creates a private S3 bucket. (TODO: compliant with NIST SP 800-53 controls)
- */
-export class PrivateBucket extends s3.Bucket {
-  constructor(scope: cdk.Construct, id: string, props?: s3.BucketProps) {
-    super(scope, id, {
-      ...props,
-      // Override any given props with better defaults
+export interface SecureBucketProps {
+  logTargetBucket: s3.IBucket | "self";
+  logTargetPrefix: string;
+  bucketProps?: s3.BucketProps;
+}
+
+export class SecureBucket extends cdk.Construct {
+  public readonly bucket: s3.IBucket;
+
+  constructor(scope: cdk.Construct, id: string, props: SecureBucketProps) {
+    super(scope, id);
+
+    const bucket = new s3.Bucket(this, id, {
+      ...props.bucketProps,
       publicReadAccess: false,
       blockPublicAccess: s3.BlockPublicAccess.BLOCK_ALL,
       autoDeleteObjects: false,
```

Comment:
```suggestion
      // sensible defaults allowed to be overriden
      accessControl: props.logTargetBucket === "self" ? undefined : s3.BucketAccessControl.LOG_DELIVERY_WRITE,
      // passed in configuration options
      ...props.bucketProps,
      // secure defaults that cannot be overridden
      publicReadAccess: false,
      blockPublicAccess: s3.BlockPublicAccess.BLOCK_ALL,
      autoDeleteObjects: false,
      serverAccessLogsBucket: props.logTargetBucket === "self" ? undefined : props.logTargetBucket.bucketName,
      serverAccessLogsPrefix: props.logTargetPrefix,
```

Reply:
Added suggested changes with small modifications to the final values for `accessControl` and `serverAccessLogsBucket`. Swapped the valves for `accessControl` since `LOG_DELIVERY_WRITE` was being applied to all buckets other than when "self" is assigned to the `logTargetBucket` property. Removed `.bucketName`.